documentation/environment-variables.md

```diff
@@ -12,11 +12,11 @@ The frontend [React](https://reactjs.org/) projects uses environment variables i
 
 ### Build-Time Variables
 
-The **apps** subprojects (./) have `.env` files that are copied from `.env.example` template files like the Laravel project. When the app projects are built using [webpack](https://webpack.js.org/), variables from the `.env` file are defined in the resulting bundle using [DefinePlugin](https://webpack.js.org/plugins/define-plugin/). These values are "baked into" the bundle and if there are any variable changes, the subproject must be rebuilt to reflect them.
+The **apps** subprojects (./) have `.env` files that are copied from `.env.example` template files like the Laravel project. When the app projects are built using [vite](https://vitejs.dev/), variables from the `.env` file are defined in the resulting bundle using [define option](https://vitejs.dev/config/shared-options.html#define). These values are "baked into" the bundle and if there are any variable changes, the subproject must be rebuilt to reflect them.
 
 ### Run-Time Variables
 
-For deployment in production, there needs to be a way to change variables in the program without rebuilding. This allows us to use environment variables as "feature flags", which enable/disable specific features with the same deployed code artifact. This is done using [envsubst](https://www.gnu.org/software/gettext/manual/html_node/envsubst-Invocation.html) to fill environment variables into the `config.js` files while setting up the app service, post-deploy. In a development environment, these variables are injected into the `config.js` files by webpack during build. To update these variables locally, just update the `.env` file and rebuild the bundle.
+For deployment in production, there needs to be a way to change variables in the program without rebuilding. This allows us to use environment variables as "feature flags", which enable/disable specific features with the same deployed code artifact. This is done using [envsubst](https://www.gnu.org/software/gettext/manual/html_node/envsubst-Invocation.html) to fill environment variables into the `config.js` files while setting up the app service, post-deploy. In a development environment, these variables are injected into the `index.html` file by vite during build. To update these variables locally, just update the `.env` file and rebuild the bundle.
```

Comment:
config.js is gone now, right?

Reply:
I guess this subsequent change is TBD once the deployment process is updated.

Reply:
Yup! Will that cause any issues I am unaware of?

Reply:
I don't think so.  We should be able to call envsubst on the index.html file the same way we do currently on the config.js file.

One enhancement I've been thinking about is that we can get rid of all the "filterUnusable" nonsense if we can emit two index.html files - one with the process.env substitutions completed for local dev and one with the variables still there for deployment.  That might be more than you want to deal with right now, though.

Reply:
Yeah, that sounds possible. I think we could do something with that using the [template option](https://github.com/vbenjs/vite-plugin-html#:~:text=entry%20file%20path-,template,-string) in the html plugin along with the [modes](https://vitejs.dev/guide/env-and-mode.html#modes) from vite :thinking: 

```ts
createHtmlPlugin({
  template: modes === "production" ? "prod.index.html" : "dev.index.html"
  // ...
})
```

Reply:
That's cool.  I don't want to blow out the scope though.  I'll let you decided whether it's worth trying for or consider it later.

Reply:
Without a config file to copy, you'll need to update the post-deploy script.  I know you don't have access to Azure but the web container should be a pretty good facsimile.
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/5f53c5ae-7963-4d55-8c8a-5f2cb4d174c6)

Reply:
I think I got it in: [remove references to config.js](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10611/commits/cc73ca5bbb105d7164b8ef5d68a4c71402e9b052) :sweat_smile:

Reply:
Alright, I thanks for walking me through that. I think it should be good after [fix post deploy env substitution](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10611/commits/b74f6b515a37b9569ad3db68874cf4361a7b4e21)